SnowflakeOfflineStoreConfig import fails in the python feature server

## Expected Behavior 
Configuring snowflake does not make the AWS python server fail

## Current Behavior
```
[ERROR] FeastModuleImportError: Could not import module 'feast.infra.offline_stores.snowflake' 
while attempting to load class 'SnowflakeOfflineStoreConfig'
```

## Steps to reproduce
- Configure a snowflake offline store
- Deploy the AWS python feature server
- Query /get-online-features
### Specifications

- Version: v0.20.0
- Platform: 
- Subsystem: Python feature server AWS

## Possible Solution

--- a/sdk/python/feast/infra/feature_servers/aws_lambda/Dockerfile
+++ b/sdk/python/feast/infra/feature_servers/aws_lambda/Dockerfile
@@ -10,6 +10,8 @@ COPY go go
 COPY README.md README.md
 
 # Install Feast for AWS with Lambda dependencies
+# TODO(felixwang9817): Remove Snowflake dependencies once lazy loading of offline stores is supported.
+# See https://github.com/feast-dev/feast/issues/2566 for more details.
 RUN pip3 install -e 'sdk/python[aws,redis,snowflake]'
 RUN pip3 install -r sdk/python/feast/infra/feature_servers/aws_lambda/requirements.txt --target "${LAMBDA_TASK_ROOT}"
 

--- a/sdk/python/feast/infra/feature_servers/aws_lambda/Dockerfile
+++ b/sdk/python/feast/infra/feature_servers/aws_lambda/Dockerfile
@@ -10,7 +10,7 @@ COPY go go
 COPY README.md README.md
 
 # Install Feast for AWS with Lambda dependencies
-RUN pip3 install -e 'sdk/python[aws,redis]'
+RUN pip3 install -e 'sdk/python[aws,redis,snowflake]'
 RUN pip3 install -r sdk/python/feast/infra/feature_servers/aws_lambda/requirements.txt --target "${LAMBDA_TASK_ROOT}"
 
 # Set the CMD to your handler (could also be done as a parameter override outside of the Dockerfile)

